src/stores/AuthStore.js

```diff
@@ -25,36 +27,86 @@ const setDAOHeaders = (token: string) => {
   ]);
 };
 
+type AuthState = {|
+  roles: ?Array<string>,
+  token: ?string,
+  userName: string,
+|};
+
+const initialState = {
+  roles: null,
+  token: null,
+  userName: null,
+};
+
 class AuthStore {
-  store: RootStore;
+  _rootStore: RootStore;
 
-  @observable authState = new Map();
+  @observable authState: AuthState = initialState;
+  @observable isInitialized: boolean = false;
 
   constructor(rootStore) {
-    this.store = rootStore;
-
-    // todo move from constructor
+    this._rootStore = rootStore;
+    // todo may be it better to use reaction here
+    // it may help to avoid isInitialized prop, but I'm not sure
     autorun(() => {
-      if (this.store.errorStore.errorStatus === 401) {
-        this.clearAuthState();
+      if (!this.isInitialized) {
+        return;
+      }
+      if (this.isAuthorized) {
+        NavigationService.navigate('home');
+      } else {
+        NavigationService.navigate('login');
       }
     });
   }
 
   @action
   clearAuthState = () => {
-    this.authState.clear();
+    this.authState = initialState;
+  };
+
+  @action
+  initialize = async () => {
+    const authStateString = await AsyncStorage.getItem('authState');
```

Comment:
for state peristance, instead of do it manually, we can use several options, like:
https://github.com/farwayer/mobx-decorators#save
or
https://github.com/pinqy520/mobx-persist
both looks fine for me

Reply:
Yeah, that seems good. I like the first one better but I don't like the fact that it installs all the other ones